llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -603,6 +595,13 @@ void InterBlockScheduling::enterRegion(MachineBasicBlock *BB,
                       << "\n");
   if (!BS.FixPoint.NumIters) {
     BS.addRegion(BB, RegionBegin, RegionEnd);
+    if (BS.Kind == BlockType::Loop) {
+      // This is the first time we schedule this loop. In that first
+      // iteration, we have recorded the region decomposition.
+      // Now we can create the interblock edges between the top and the bottom
+      // region
+      BS.initInterBlock(*Context, *HR);
+    }
```

Comment:
I don't think this makes sense for a multi-region loop. We will call the interblock init for each region in the block.

Reply:
Hmm, you're completely right. We indeed need a way to gather all regions before starting the actual scheduling. I'll work on that, maybe a new `GatheringRegions` stage in our sate machine.

Reply:
sate machine. Sounds delicious.